marlin/src/marlin/marlin.rs

```diff
@@ -606,16 +628,19 @@ impl<
         }
 
         let padded_public_input = {
-            let domain_x = EvaluationDomain::<TargetField>::new(public_input.len() + 1).unwrap();
+            let input_domain = EvaluationDomain::<TargetField>::new(public_input.len() + 1).unwrap();
 
             if cfg!(debug_assertions) {
                 println!("Number of given public inputs: {}", public_input.len());
-                println!("Size of evaluation domain x: {}", domain_x.size());
+                println!("Size of evaluation domain x: {}", input_domain.size());
             }
 
             let mut new_input = vec![TargetField::one()];
             new_input.extend_from_slice(public_input);
-            new_input.resize(core::cmp::max(public_input.len(), domain_x.size()), TargetField::zero());
+            new_input.resize(
+                core::cmp::max(public_input.len(), input_domain.size()),
+                TargetField::zero(),
+            );
             assert!(new_input.first().unwrap().is_one());
```

Comment:
this assertion doesn't seem useful - `new_input[0]` is set to `TargetField::one()` right above, and it doesn't get mutated in between

Reply:
The `resize` could truncate it to zero, right?